proposals/Level-1.md

```diff
@@ -265,25 +229,30 @@ else
 end
 ```
 
-The conditional query of an exception succeeds when the exception value on the
-top of the stack has the corresponding exception tag (defined by `except_index`).
+In the first form, the instructions between the `if_except` and 'end' define the
+`then block`. In the second form, the instructions between the `if_except` and
+`else` define the `then block`, while the instructions between the `else` and
+the `end define the `else block`.
+
+The conditional query of an exception checks the exception tag of exception on
+top of the stack. It succeeds only if the exception index of the instruction
+matches the corresponding exception tag. Once the query completes, the exception
+is poppod off the stack.
 
-If the query succeeds, the values (associated with the exception value) are
+If the query succeeds the values (associated with the popped exception) are
 extracted and pushed onto the stack, and control transfers to the instructions
 in the then block.
 
 If the query fails, it either enters the else block, or transfer control to the
 end of the if_except block if there is no else block.
 
-### Debugging
+### Stack traces
 
-Earlier discussion implied that when an exception is thrown, the runtime will
-pop the operand stack across function calls until a corresponding, enclosing try
-block is found. The implementation may actually not do this. Rather, it may
-first search up the call stack to see if there is an enclosing try. If none are
-found, it could terminate the thread at the point of the throw. This would
-allow better debugging capability, since the corresponding call stack is still
-there to query.
+When an exception is thrown, the runtime will pop the operand stack across
+function calls until a corresponding, enclosing try block is found. It may also
+associate a stack trace that can be used to report uncaught exceptions. However,
+the details of this is left to the embedder, and WebAssembly does not have
+(direct) access to the stack trace.
```

Comment:
1 - The stack trace is intended to be a call stack trace, and my read still states this.

2. We may have two phase (for debugging), but we never stated that it would be mandatory. Dropped because that is an implementation detail and not a requirement.

Reply:
For what it's worth, stack traces are also an implementation detail since there is no way to access them from Wasm.

Reply:
Removed the end of the last sentence (on direct access), since it is optional of the implementation.